app/src/main/res/layout/activity_main.xml

```diff
@@ -35,33 +37,62 @@
         android:id="@+id/server_button"
         android:layout_width="wrap_content"
         android:layout_height="wrap_content"
+        android:layout_marginTop="400dp"
         android:fontFamily="@font/bitter"
         android:text="@string/behave_as_server"
         app:layout_constraintBottom_toTopOf="@id/client_button"
         app:layout_constraintEnd_toEndOf="parent"
         app:layout_constraintStart_toStartOf="parent"
-        app:layout_constraintTop_toTopOf="parent"
-        app:layout_constraintVertical_chainStyle="packed" />
+        app:layout_constraintTop_toBottomOf="@id/input_username"
+        app:layout_constraintVertical_chainStyle="spread" />
 
     <Button
         android:id="@+id/client_button"
         android:layout_width="wrap_content"
         android:layout_height="wrap_content"
-        android:layout_margin="20dp"
+        android:layout_margin="10dp"
+        android:layout_marginBottom="0dp"
         android:fontFamily="@font/bitter"
+        android:rotationY="0"
         android:text="@string/behave_as_client"
         app:layout_constraintBottom_toBottomOf="parent"
         app:layout_constraintEnd_toEndOf="parent"
         app:layout_constraintStart_toStartOf="parent"
-        app:layout_constraintTop_toBottomOf="@id/server_button" />
+        app:layout_constraintTop_toBottomOf="@+id/server_button" />
 
-    <androidx.fragment.app.FragmentContainerView
+    <androidx.constraintlayout.widget.Guideline
+        android:layout_width="wrap_content"
+        android:layout_height="wrap_content"
+        app:layout_constraintGuide_percent="0.7"
+        android:id="@+id/horizontal_guide"
+        android:orientation="horizontal"/>
+
+    <EditText
+        android:id="@+id/input_username"
+        android:layout_width="0dp"
+        android:layout_height="wrap_content"
+        android:layout_marginTop="-20dp"
+        android:hint="@string/enter_username"
+        android:inputType="text"
+        android:maxLength="5"
```

Comment:
Does `5` really make sense? Seems a bit short to me

Reply:
Was an example length provided by me - we want to avoid new lines breaking up the name,
Ex.:
Lech
  ld
Instead of:
Lechld
and there is not that much space to write the name without a newline in the current gameboard. Exact number of characters would have to be tested - but 4-5 should work.